0-module-and-module-source.md

```diff
@@ -48,33 +48,39 @@ of an agent cluster.
 ```ts
 type ImportSpecifier = string;
 
-type ImportHook = (specifier: ImportSpecifier, importMeta: object) =>
+type ImportHook = (this: ModuleHandler, specifier: ImportSpecifier) =>
   Promise<Module>;
 
+type ImportMetaHook = (this: ModuleHandler, importMeta: any) => any;
+
+type ModuleHandler = {
+  importHook?: ImportHook,
+  importMetaHook?: ImportMetaHook,
+  // ...
```

Comment:
there is a form in typescript to represent a well define object that allows expandos, just add ``

```suggestion
  [name: PropertyKey]: any
```